Tests failed with encondig set

Hi!
default tests from molecule failed 

```
def test_hosts_file(File):
    hosts = File('/etc/hosts')

    assert hosts.user == 'root'
    assert hosts.group == 'root'
```

on server with locale LANG=ru_RU.UTF-8
tests filed with

```
_______________________________________________________________ ERROR at setup of test_hosts_file[ansible://tower-01] _______________________________________________________________
[gw0] linux2 -- Python 2.7.11 /usr/bin/python
TestinfraBackend = <testinfra.backend.ansible.AnsibleBackend object at 0x7fb567387b90>

    @pytest.fixture()
    def f(TestinfraBackend):
>       return TestinfraBackend.get_module(cls.__name__)

/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:75: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python2.7/dist-packages/testinfra/backend/base.py:225: in get_module
    module = getattr(testinfra.modules, name).get_module(self)
/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:62: in get_module
    klass = cls.get_module_class(_backend)
/usr/local/lib/python2.7/dist-packages/testinfra/modules/file.py:178: in get_module_class
    if SystemInfo.type == "linux":
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:103: in type
    return self.sysinfo["type"]
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:31: in sysinfo
    self._sysinfo = self.get_system_info()
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:88: in get_system_info
    sysinfo.update(**self._get_linux_sysinfo())
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:38: in _get_linux_sysinfo
    lsb = self.run("lsb_release -a")
/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:23: in run
    return self._backend.run(command, *args, **kwargs)
/usr/local/lib/python2.7/dist-packages/testinfra/backend/ansible.py:43: in run
    stderr_bytes = b"".join((chr(ord(c)) for c in out['stderr']))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <iterator object at 0x7fb564e78f90>

>   stderr_bytes = b"".join((chr(ord(c)) for c in out['stderr']))
E   ValueError: chr() arg not in range(256)
```

i am using molecule 1.5.1 and ansible 2.1.0.0 and

```
% testinfra --version
This is pytest version 2.9.2, imported from /usr/local/lib/python2.7/dist-packages/pytest.pyc
setuptools registered plugins:
  testinfra-1.3.1 at /usr/local/lib/python2.7/dist-packages/testinfra/plugin.pyc
  pytest-xdist-1.14 at /usr/local/lib/python2.7/dist-packages/xdist/boxed.pyc
  pytest-xdist-1.14 at /usr/local/lib/python2.7/dist-packages/xdist/looponfail.pyc
  pytest-xdist-1.14 at /usr/local/lib/python2.7/dist-packages/xdist/plugin.pyc
```

that can be workarounded with explicit set export LANG=C

yep. here the full output 
```
Executing testinfra tests found in tests/.
============================= test session starts ==============================
platform linux2 -- Python 2.7.11+, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/shirokih/vagrant/molecule/tower, inifile: 
plugins: testinfra-1.3.1, xdist-1.14
gw0 [4] / gw1 [4] / gw2 [4]
scheduling tests via LoadScheduling
EEEE
==================================== ERRORS ====================================
____________ ERROR at setup of test_hosts_file[ansible://tower-01] _____________
[gw1] linux2 -- Python 2.7.11 /usr/bin/python
TestinfraBackend = <testinfra.backend.ansible.AnsibleBackend object at 0x7fd3adc84cd0>

    @pytest.fixture()
    def f(TestinfraBackend):
>       return TestinfraBackend.get_module(cls.__name__)

/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:75: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python2.7/dist-packages/testinfra/backend/base.py:225: in get_module
    module = getattr(testinfra.modules, name).get_module(self)
/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:62: in get_module
    klass = cls.get_module_class(_backend)
/usr/local/lib/python2.7/dist-packages/testinfra/modules/file.py:178: in get_module_class
    if SystemInfo.type == "linux":
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:103: in type
    return self.sysinfo["type"]
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:31: in sysinfo
    self._sysinfo = self.get_system_info()
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:88: in get_system_info
    sysinfo.update(**self._get_linux_sysinfo())
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:38: in _get_linux_sysinfo
    lsb = self.run("lsb_release -a")
/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:23: in run
    return self._backend.run(command, *args, **kwargs)
/usr/local/lib/python2.7/dist-packages/testinfra/backend/ansible.py:43: in run
    stderr_bytes = b"".join((chr(ord(c)) for c in out['stderr']))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <iterator object at 0x7fd3a76141d0>

>   stderr_bytes = b"".join((chr(ord(c)) for c in out['stderr']))
E   ValueError: chr() arg not in range(256)

/usr/local/lib/python2.7/dist-packages/testinfra/backend/ansible.py:43: ValueError
___ ERROR at setup of test_noc_tower_running_and_enabled[ansible://tower-01] ___
[gw0] linux2 -- Python 2.7.11 /usr/bin/python
TestinfraBackend = <testinfra.backend.ansible.AnsibleBackend object at 0x7f6c81a71cd0>

    @pytest.fixture()
    def f(TestinfraBackend):
>       return TestinfraBackend.get_module(cls.__name__)

/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:75: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python2.7/dist-packages/testinfra/backend/base.py:225: in get_module
    module = getattr(testinfra.modules, name).get_module(self)
/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:62: in get_module
    klass = cls.get_module_class(_backend)
/usr/local/lib/python2.7/dist-packages/testinfra/modules/service.py:50: in get_module_class
    File = _backend.get_module("File")
/usr/local/lib/python2.7/dist-packages/testinfra/backend/base.py:225: in get_module
    module = getattr(testinfra.modules, name).get_module(self)
/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:62: in get_module
    klass = cls.get_module_class(_backend)
/usr/local/lib/python2.7/dist-packages/testinfra/modules/file.py:178: in get_module_class
    if SystemInfo.type == "linux":
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:103: in type
    return self.sysinfo["type"]
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:31: in sysinfo
    self._sysinfo = self.get_system_info()
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:88: in get_system_info
    sysinfo.update(**self._get_linux_sysinfo())
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:38: in _get_linux_sysinfo
    lsb = self.run("lsb_release -a")
/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:23: in run
    return self._backend.run(command, *args, **kwargs)
/usr/local/lib/python2.7/dist-packages/testinfra/backend/ansible.py:43: in run
    stderr_bytes = b"".join((chr(ord(c)) for c in out['stderr']))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <iterator object at 0x7f6c82f3dbd0>

>   stderr_bytes = b"".join((chr(ord(c)) for c in out['stderr']))
E   ValueError: chr() arg not in range(256)

/usr/local/lib/python2.7/dist-packages/testinfra/backend/ansible.py:43: ValueError
_________ ERROR at setup of test_deploy_keys_setup[ansible://tower-01] _________
[gw2] linux2 -- Python 2.7.11 /usr/bin/python
TestinfraBackend = <testinfra.backend.ansible.AnsibleBackend object at 0x7fefc00a9cd0>

    @pytest.fixture()
    def f(TestinfraBackend):
>       return TestinfraBackend.get_module(cls.__name__)

/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:75: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python2.7/dist-packages/testinfra/backend/base.py:225: in get_module
    module = getattr(testinfra.modules, name).get_module(self)
/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:62: in get_module
    klass = cls.get_module_class(_backend)
/usr/local/lib/python2.7/dist-packages/testinfra/modules/file.py:178: in get_module_class
    if SystemInfo.type == "linux":
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:103: in type
    return self.sysinfo["type"]
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:31: in sysinfo
    self._sysinfo = self.get_system_info()
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:88: in get_system_info
    sysinfo.update(**self._get_linux_sysinfo())
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:38: in _get_linux_sysinfo
    lsb = self.run("lsb_release -a")
/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:23: in run
    return self._backend.run(command, *args, **kwargs)
/usr/local/lib/python2.7/dist-packages/testinfra/backend/ansible.py:43: in run
    stderr_bytes = b"".join((chr(ord(c)) for c in out['stderr']))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <iterator object at 0x7fefb98471d0>

>   stderr_bytes = b"".join((chr(ord(c)) for c in out['stderr']))
E   ValueError: chr() arg not in range(256)

/usr/local/lib/python2.7/dist-packages/testinfra/backend/ansible.py:43: ValueError
___________ ERROR at setup of test_db_file_setup[ansible://tower-01] ___________
[gw1] linux2 -- Python 2.7.11 /usr/bin/python
TestinfraBackend = <testinfra.backend.ansible.AnsibleBackend object at 0x7fd3adc84cd0>

    @pytest.fixture()
    def f(TestinfraBackend):
>       return TestinfraBackend.get_module(cls.__name__)

/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:75: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python2.7/dist-packages/testinfra/backend/base.py:225: in get_module
    module = getattr(testinfra.modules, name).get_module(self)
/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:62: in get_module
    klass = cls.get_module_class(_backend)
/usr/local/lib/python2.7/dist-packages/testinfra/modules/file.py:178: in get_module_class
    if SystemInfo.type == "linux":
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:103: in type
    return self.sysinfo["type"]
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:31: in sysinfo
    self._sysinfo = self.get_system_info()
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:88: in get_system_info
    sysinfo.update(**self._get_linux_sysinfo())
/usr/local/lib/python2.7/dist-packages/testinfra/modules/systeminfo.py:38: in _get_linux_sysinfo
    lsb = self.run("lsb_release -a")
/usr/local/lib/python2.7/dist-packages/testinfra/modules/base.py:23: in run
    return self._backend.run(command, *args, **kwargs)
/usr/local/lib/python2.7/dist-packages/testinfra/backend/ansible.py:43: in run
    stderr_bytes = b"".join((chr(ord(c)) for c in out['stderr']))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <iterator object at 0x7fd3a75fef90>

>   stderr_bytes = b"".join((chr(ord(c)) for c in out['stderr']))
E   ValueError: chr() arg not in range(256)

/usr/local/lib/python2.7/dist-packages/testinfra/backend/ansible.py:43: ValueError
=========================== 4 error in 2.79 seconds ============================
ERROR: 

  RAN: '/usr/local/bin/testinfra tests/test_default.py tests/test_env.py -n 3 --connection=ansible --ansible-inventory=.molecule/ansible_inventory'

  STDOUT:


  STDERR:

[shirokih-hom] 15:25:27 /home/shirokih/vagrant/molecule/tower % lsb_release -a
No LSB modules are available.
Distributor ID:	LinuxMint
Description:	Linux Mint 18 Sarah
Release:	18
Codename:	sarah
[shirokih-hom] 15:26:27 /home/shirokih/vagrant/molecule/tower % locale
LANG=ru_RU.UTF-8
LANGUAGE=
LC_CTYPE="ru_RU.UTF-8"
LC_NUMERIC=en_US.UTF-8
LC_TIME=ru_RU.UTF-8
LC_COLLATE="ru_RU.UTF-8"
LC_MONETARY=en_US.UTF-8
LC_MESSAGES="ru_RU.UTF-8"
LC_PAPER=en_US.UTF-8
LC_NAME=en_US.UTF-8
LC_ADDRESS=en_US.UTF-8
LC_TELEPHONE=en_US.UTF-8
LC_MEASUREMENT=en_US.UTF-8
LC_IDENTIFICATION=en_US.UTF-8
LC_ALL=
[shirokih-hom] 15:26:41 /home/shirokih/vagrant/molecule/tower % molecule login
Warning: Permanently added '[127.0.0.1]:2222' (ECDSA) to the list of known hosts.
Last login: Wed Jun 29 08:25:23 2016 from 10.0.2.2
[vagrant@localhost ~]$ lsb_release -a
-bash: lsb_release: command not found
[vagrant@localhost ~]$ sudo -s
[root@localhost vagrant]# lsb_release -a
bash: lsb_release: command not found
[root@localhost vagrant]# exit
[vagrant@localhost ~]$ logout
Connection to 127.0.0.1 closed.
```

--- a/testinfra/backend/ansible.py
+++ b/testinfra/backend/ansible.py
@@ -43,21 +43,26 @@ class AnsibleBackend(base.BaseBackend):
         command = self.get_command(command, *args)
         out = self.run_ansible("shell", module_args=command)
 
-        # Ansible return an unicode object but this is bytes ...
+        # Ansible may return bytes as an unicode object...
         # A simple test case is:
         # >>> assert File("/bin/true").content == open("/bin/true").read()
-        stdout_bytes = b"".join((chr(ord(c)) for c in out['stdout']))
-        stderr_bytes = b"".join((chr(ord(c)) for c in out['stderr']))
+        try:
+            stdout_bytes = b"".join((chr(ord(c)) for c in out['stdout']))
+        except ValueError:
+            stdout_bytes = None
 
-        result = base.CommandResult(
-            self, out['rc'],
-            stdout_bytes,
-            stderr_bytes,
+        try:
+            stderr_bytes = b"".join((chr(ord(c)) for c in out['stderr']))
+        except ValueError:
+            stderr_bytes = None
+
+        return self.result(
+            out['rc'],
             command,
+            stdout_bytes=stdout_bytes,
+            stderr_bytes=stderr_bytes,
             stdout=out["stdout"], stderr=out["stderr"],
         )
-        logger.info("RUN %s", result)
-        return result
 
     def run_ansible(self, module_name, module_args=None, **kwargs):
         result = self.ansible_runner.run(

--- a/testinfra/backend/base.py
+++ b/testinfra/backend/base.py
@@ -26,12 +26,12 @@ logger = logging.getLogger("testinfra")
 class CommandResult(object):
 
     def __init__(
-        self, backend, exit_status, stdout_bytes, stderr_bytes, command,
-        stdout=None, stderr=None,
+        self, backend, exit_status, command, stdout_bytes,
+        stderr_bytes, stdout=None, stderr=None,
     ):
         self.exit_status = exit_status
-        self.stdout_bytes = stdout_bytes
-        self.stderr_bytes = stderr_bytes
+        self._stdout_bytes = stdout_bytes
+        self._stderr_bytes = stderr_bytes
         self._stdout = stdout
         self._stderr = stderr
         self.command = command
@@ -45,14 +45,26 @@ class CommandResult(object):
     @property
     def stdout(self):
         if self._stdout is None:
-            self._stdout = self._backend.decode(self.stdout_bytes)
+            self._stdout = self._backend.decode(self._stdout_bytes)
         return self._stdout
 
     @property
     def stderr(self):
         if self._stderr is None:
-            self._stderr = self._backend.decode(self.stderr_bytes)
+            self._stderr = self._backend.decode(self._stderr_bytes)
         return self._stderr
+
+    @property
+    def stdout_bytes(self):
+        if self._stdout_bytes is None:
+            self._stdout_bytes = self._backend.encode(self._stdout)
+        return self._stdout_bytes
+
+    @property
+    def stderr_bytes(self):
+        if self._stderr_bytes is None:
+            self._stderr_bytes = self._backend.encode(self._stderr)
+        return self._stderr_bytes
 
     def __repr__(self):
         return (
@@ -61,8 +73,8 @@ class CommandResult(object):
         ) % (
             repr(self.command),
             self.exit_status,
-            repr(self.stdout_bytes),
-            repr(self.stderr_bytes),
+            repr(self._stdout_bytes or self._stdout),
+            repr(self._stderr_bytes or self._stderr),
         )
 
 
@@ -160,10 +172,7 @@ class BaseBackend(object):
             stderr=subprocess.PIPE,
         )
         stdout, stderr = p.communicate()
-        result = CommandResult(
-            self, p.returncode, stdout, stderr, command,
-        )
-        logger.info("RUN %s", result)
+        result = self.result(p.returncode, command, stdout, stderr)
         return result
 
     @staticmethod
@@ -205,6 +214,11 @@ class BaseBackend(object):
             return data.encode("ascii")
         except UnicodeEncodeError:
             return data.encode(self.encoding)
+
+    def result(self, *args, **kwargs):
+        result = CommandResult(self, *args, **kwargs)
+        logger.info("RUN %s", result)
+        return result
 
     def get_module(self, name):
         """Return the testinfra module adapted to the current backend

--- a/testinfra/backend/paramiko.py
+++ b/testinfra/backend/paramiko.py
@@ -14,7 +14,6 @@
 from __future__ import unicode_literals
 from __future__ import absolute_import
 
-import logging
 import os
 
 try:
@@ -31,7 +30,6 @@ else:
             pass
 
 from testinfra.backend import base
-logger = logging.getLogger("testinfra")
 
 
 class ParamikoBackend(base.BaseBackend):
@@ -99,6 +97,4 @@ class ParamikoBackend(base.BaseBackend):
             else:
                 raise
 
-        result = base.CommandResult(self, rc, stdout, stderr, command)
-        logger.info("RUN %s", result)
-        return result
+        return self.result(rc, command, stdout, stderr)

--- a/testinfra/backend/salt.py
+++ b/testinfra/backend/salt.py
@@ -14,8 +14,6 @@
 from __future__ import unicode_literals
 from __future__ import absolute_import
 
-import logging
-
 try:
     import salt.client
 except ImportError:
@@ -24,7 +22,6 @@ else:
     HAS_SALT = True
 
 from testinfra.backend import base
-logger = logging.getLogger("testinfra")
 
 
 class SaltBackend(base.BaseBackend):
@@ -52,10 +49,8 @@ class SaltBackend(base.BaseBackend):
     def run(self, command, *args):
         command = self.get_command(command, *args)
         out = self.run_salt("cmd.run_all", [command])
-        result = base.CommandResult(
-            self, out['retcode'], out['stdout'], out['stderr'], command)
-        logger.info("RUN %s", result)
-        return result
+        return self.result(out['retcode'], command, out['stdout'],
+                           out['stderr'])
 
     def run_salt(self, func, args=None):
         out = self.client.cmd(self.host, func, args or [])

--- a/testinfra/backend/ssh.py
+++ b/testinfra/backend/ssh.py
@@ -80,6 +80,4 @@ class SafeSshBackend(SshBackend):
         rc = int(rc)
         stdout = base64.b64decode(stdout)
         stderr = base64.b64decode(stderr)
-        return base.CommandResult(
-            self, rc, stdout, stderr,
-            self.encode(orig_command))
+        return self.result(rc, self.encode(orig_command), stdout, stderr)

